cpp/tests/groupby/groupby_test_util.hpp

```diff
@@ -73,6 +73,12 @@ inline void test_single_agg(column_view const& keys,
                             std::vector<order> const& column_order = {},
                             std::vector<null_order> const& null_precedence = {})
 {
+  auto const sort_expect_order = sorted_order(table_view{{expect_keys}}, {}, {null_order::AFTER});
+  auto const sorted_expect_keys =
+    gather(table_view{{expect_keys}}, *sort_expect_order)->view().column(0);
+  auto const sorted_expect_vals =
+    gather(table_view{{expect_vals}}, *sort_expect_order)->view().column(0);
```

Comment:
@divyegala The problem is here. `gather` returns a unique ptr of `cudf::table` thus you are dereferencing a temporary ptr...
```suggestion
  auto const sorted_expect_keys =
    gather(table_view{{expect_keys}}, *sort_expect_order);
  auto const sorted_expect_vals =
    gather(table_view{{expect_vals}}, *sort_expect_order);
```

Reply:
Hah, I didn't notice that at the first time too :stuck_out_tongue:

Reply:
Why do we return a `unique_ptr`, btw? I imagine this is from C++11/14 but if either it is not trivially copyable/copy constructor is deleted, why don't we rely on copy-elision to return objects?

Reply:
For your reference:

- https://github.com/rapidsai/cudf/blob/branch-23.02/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#views-and-ownership
- https://github.com/rapidsai/cudf/blob/branch-23.02/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#inputoutput-style